modules/core/src/utils/flatten.js

```diff
@@ -142,3 +142,85 @@ export function flattenTypedVertices(nestedArray, {
   return start;
 }
 */
+
+/*
+ * Helper function for padArray
+ */
+function padSubArray({source, target, start = 0, end, enter}) {
+  end = end || target.length;
+
+  const sourceLength = source.length;
+  const targetLength = end - start;
+
+  if (sourceLength > targetLength) {
+    target.set(source.subarray(0, targetLength), start);
```

Comment:
Thought: Should we call our own `copyArray` function, that internally uses `set` or iteration based on best performance characteristics for the length of the data?